docs/src/views/Layout.tsx

```diff
@@ -0,0 +1,212 @@
+        )
+      `}
+    />
+    <ExampleSnippet
+      render={() => {
+        const OpaqueBackground = ({ children, color }) => (
+          <Segment styles={{ backgroundColor: color }} content={{ children }} />
+        )
+
+        return (
+          <OpaqueBackground color="red">
+            This is a bad approach to opaque background :(
+          </OpaqueBackground>
+        )
+      }}
+    />
+    <ExampleSnippet
+      render={() => {
+        const RelativePositioned = ({ children, left }) => (
+          <Segment styles={{ position: 'relative', left }} content={children} />
+        )
+
+        return (
+          <RelativePositioned left={'30px'}>
+            This is a bad way to support relative-positioning :(
+          </RelativePositioned>
+        )
+      }}
+    />
+    <p>
+      While it might seem that the intent is addressed with the approach taken, however, this is
+      wrong for the following reason:
+    </p>
+    <blockquote>
+      Stardust makes it very important to{' '}
+      <strong>follow component's semantics at the first place</strong>, and only then consider the
+      visual aspects.
+    </blockquote>
+    <p>If we'd refer to the semantics provided in the description for {links.segment} component:</p>
+    <blockquote>A segment is used to create a grouping of related content.</blockquote>
+    <p>
+      This description suggests that purpose of {code('Segment')} is not a layout. Thus, this
+      component shouldn't be used for layout purposes, as this will break component's semantics -
+      and, as a consequence, it might break accessibility and theming, as those both adhere to
+      component's semantics at the first place.
+    </p>
+    <Divider />
+    <Header as="h2">Flex recipies</Header>
+    There are several advices that might help when using {code('Flex')} component.
+    <Header as="h3">{code('Flex.Item')} should be used sparingly</Header>
+    <blockquote>
+      Use {code('Flex.Item')} component to wrap child elements only in case if their flex styles
+      should be overriden.
+    </blockquote>
+    <p>
+      There is no strict need to use {code('Flex.Item')} component as a direct child of{' '}
+      {code('Flex')} - one may use {code('Flex.Item')} component only when it is necessary to tweak
+      flex styles of individual child item.
+    </p>
+    <ExampleSnippet
+      render={() => (
+        <Flex gap="gap.small">
+          <Button content="Accept" />
+          <Button content="Deny" />
+        </Flex>
+      )}
+    />
+    <p>As a consequence of this:</p>
+    <blockquote>
+      Each {code('Flex.Item')} element being introduced without any props specified should be
+      considered for removal.
+    </blockquote>
+    <ExampleSnippet
+      render={() => (
+        <Flex gap="gap.small">
+          <Button content="Accept" />
+          <Flex.Item>
+            <Button content="Deny" />
+          </Flex.Item>
+        </Flex>
```

Comment:
I like the way how it's done on `overreacted.io`:
![image](https://user-images.githubusercontent.com/14183168/54829721-af002e80-4cbf-11e9-8164-c9117b627266.png)

---

![image](https://user-images.githubusercontent.com/14183168/54829940-2fbf2a80-4cc0-11e9-9ddb-ad46bcfac0ce.png)

Something like this will be really nice :)

Reply:
yes, fully agree - I would say, anything that would allow us to have comments preserved would be already a huge win :) for now can do either of two:
- use the approach currently taken
- or write code twice - one for example's code, one for the actual rendering

Given these options, have chosen the first approach because, essentially, it will at least guarantee that these two things (code and rendered output) won't go out of sync

Reply:
Let's leave as is for now, duplication is more evil 😈